rust/windows-client/src-tauri/src/client/logging.rs

```diff
@@ -72,33 +72,50 @@ pub(crate) async fn clear_logs_inner() -> Result<()> {
 pub(crate) async fn export_logs_inner(ctlr_tx: CtlrTx) -> Result<()> {
     let now = chrono::Local::now();
     let datetime_string = now.format("%Y_%m_%d-%H-%M");
-    let filename = format!("connlib-{datetime_string}.zip");
+    let stem = format!("connlib-{datetime_string}");
+    let filename = format!("{stem}.zip");
 
     tauri::api::dialog::FileDialogBuilder::new()
         .add_filter("Zip", &["zip"])
         .set_file_name(&filename)
         .save_file(move |file_path| match file_path {
             None => {}
-            Some(x) => {
+            Some(path) => {
                 // blocking_send here because we're in a sync callback within Tauri somewhere
                 ctlr_tx
-                    .blocking_send(ControllerRequest::ExportLogs(x))
+                    .blocking_send(ControllerRequest::ExportLogs { path, stem })
                     .unwrap()
             }
         });
     Ok(())
 }
 
 /// Exports logs to a zip file
-pub(crate) async fn export_logs_to(file_path: PathBuf) -> Result<()> {
-    tracing::trace!("Exporting logs to {file_path:?}");
+/// To avoid the ["tar bomb"](https://www.linfo.org/tarbomb.html) problem, all files
+/// are put into a subdirectory with the automatically-generated file name of the zip,
+/// even if the user customized the zip's file name.
+pub(crate) async fn export_logs_to(path: PathBuf, stem: String) -> Result<()> {
+    tracing::trace!("Exporting logs to {path:?}");
+
+    spawn_blocking(move || {
+        let f = fs::File::create(path)?;
+        let mut zip = zip::ZipWriter::new(f);
+        // All files will have the same modified time. Doing otherwise seems to be difficult
+        let options = zip::write::FileOptions::default();
+        for entry in fs::read_dir("logs")? {
+            let entry = entry?;
+            let name = entry.file_name();
+            let name = name.to_string_lossy();
+            // TODO: Before merging, don't do sync file I/O in an async fn
+            zip.start_file(format!("{stem}/{name}"), options)?;
```

Comment:
we should use [`Path`'s methods](https://doc.rust-lang.org/std/path/struct.Path.html#method.join) for things like this as it's less error prone.